.travis.yml

```diff
@@ -0,0 +1,29 @@
+language: java
+
+# This project is aimed to be JDK7-compatible; however, there are
+# issues with testing against it.
+# openjdk7: https://github.com/gradle/gradle/issues/2421
```

Comment:
Could you try the workaround listed for `openjdk7` ~ https://github.com/gradle/gradle/issues/2421#issuecomment-319916874

Reply:
Tried. Unfortunately, ran into another issue (see [here](https://travis-ci.org/uber-java/tally/jobs/272236748)) where checkstyle was basically complaining that the JRE is running a different Java version that the JDK. The workaround basically compiles using JDK8 and runs the rest using JRE7, which does not give us much extra.

Reply:
> The workaround basically compiles using JDK8 and runs the rest using JRE7, which does not give us much extra.

Isn't that sufficient for us to maintain JRE7 compatibility?

Reply:
Technically, yes, although this seems rather hacky as this is not supported and I'm not sure if there are unintended effects of running in this inconsistent mode.

> Class files built with the Java SE 8 compiler will not run on earlier releases of Java SE.

http://www.oracle.com/technetwork/java/javase/8-compatibility-guide-2156366.html#A999081

Reply:
As discussed offline, I changed the build to use the latest checkstyle version supporting JRE7 so that we can run tests against JRE7.